### PR TITLE
feat(logs): allow setting max log size

### PR DESCRIPTION
**Pull request by wass3r**
adds a new `VELA_EXECUTOR_MAX_LOG_SIZE` flag to limit the maximum log size. the worker will halt uploading logs at that point.

a value of `0` (default) will **not** limit the log size (that's the current behavior).

closes https://github.com/go-vela/community/issues/447

### CODE REVIEW

File: executor/linux/service.go [hunk 1]
```diff
@@ -197,6 +197,13 @@ func (c *client) StreamService(ctx context.Context, ctn *pipeline.Container) err
 			return
 		}
 
+		// don't attempt last upload if log size exceeded
+		if c.maxLogSize > 0 && uint(len(data)) >= c.maxLogSize {
+			logger.Trace("maxmimum log size reached")
```
Comment (KellyMerrick) [inline executor/linux/service.go:202]:
```suggestion
			logger.Trace("maximum log size reached")
```

File: executor/linux/step.go [hunk 2]
```diff
@@ -232,6 +232,13 @@ func (c *client) StreamStep(ctx context.Context, ctn *pipeline.Container) error
 			return
 		}
 
+		// don't attempt last upload if log size exceeded
+		if c.maxLogSize > 0 && uint(len(data)) >= c.maxLogSize {
+			logger.Trace("maxmimum log size reached")
```
Comment (KellyMerrick) [inline executor/linux/step.go:237]:
```suggestion
			logger.Trace("maximum log size reached")
```

**Review by KellyMerrick — CHANGES_REQUESTED** [review]
A couple spelling errors to correct.

**Comment by wass3r** [comment]
should be fixed in latest commit - thanks @KellyMerrick

File: executor/linux/step_test.go [hunk 3]
```diff
@@ -323,6 +323,10 @@
 	_build := testBuild()
 	_repo := testRepo()
 	_user := testUser()
+	_logs := new(library.Log)
+
+	// fill log with bytes
+	_logs.SetData(make([]byte, 1000, 1000))
```
[the remark this answers is not in the document]
Comment (wass3r) [reply executor/linux/step_test.go:329]: oh you..

File: executor/flags.go [hunk 4]
```diff
@@ -49,4 +49,10 @@ var Flags = []cli.Flag{
 		Usage:    "method used to publish logs to the server - options: (byte-chunks|time-chunks)",
 		Value:    "byte-chunks",
 	},
+	&cli.UintFlag{
+		EnvVars:  []string{"VELA_EXECUTOR_MAX_LOG_SIZE", "EXECUTOR_MAX_LOG_SIZE"},
+		FilePath: "/vela/executor/max_log_size",
+		Name:     "executor.max_log_size",
+		Usage:    "maximum log size (in bytes)",
+	},
```
Comment (jbrockopp) [inline executor/flags.go:57]: @wass3r could you help me understand why we're using a `uint` here?

Comment (wass3r) [reply executor/flags.go:57]: because negative integers shouldn't be permitted. saves some LOCs on error checking.

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by KellyMerrick — APPROVED** [review]

**Pull request merged**